bus/protocol.c

```diff
@@ -407,18 +411,22 @@ int gip_request_paddles_info(struct gip_client *client) {
     /* always fails on adapter removal */
     err = adap->ops->submit_buffer(adap, &buf);
     if (err)
-        dev_dbg(&client->dev, "%s: submit paddle request buffer failed: %d\n",
-                __func__, err);
+    {
+        dev_dbg(&client->dev,
+                "%s: submit paddle request buffer failed: %d\n",
+                __func__,
+                err);
+    }
 
 err_unlock:
```

Comment:
If you don't like using `goto`, you could move most of these codes into a helper function, like this:

```c
int helper(clint* client)
{
    if(err)
    {
        // do something
        return err;
    }
    if(invalid_length())
    {
        // do something
        return err;
    }
    // do something
    return err;
}

int gip_request_paddles_info(struct gip_client *client)
{
    int err = helper(client);
    spin_unlock_irqrestore(&adap->send_lock, flags);
    return err;
}
```